person.rb

```diff
@@ -0,0 +1,38 @@
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :name, :age, :rentals
+  attr_reader :id
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def generate_id
+    rand(1..1000)
+  end
+
+  def add_rental(rental)
+    @rentals.push
+    rental.person = self
+  end
```

Comment:
When you created the `rental` class, in the `constructor`, you add the rental object to both the `person`'s rentals array and the `book`'s rentals array via the following lines: 
```
 person.rentals << self
 book.rentals << self
```
  So, it would make more sense if instead of passing a rental object as a parameter here, you pass the `date` and the `book`, and in the body of the method, you create a new rental (To create a new rental, we also need a `person`, for that, you can use `self`).  This way, the constructor will take care of adding it to the list of rentals for both the book and the person.

Something like this will do the trick:
```
   def add_rental(book, date)
       Rental.new(date, book, self)
   end

```